app/models/post.rb

```diff
@@ -1,17 +1,18 @@
 class Post < ApplicationRecord
+  after_save :update_post_counter
   belongs_to :author, class_name: 'User'
   has_many :comments
   has_many :likes
 
   validates :title, presence: true, length: { maximum: 250 }
-  validates :comments_counter, numericality: { greater_than_or_equal_to: 0 }
-  validates :likes_counter, numericality: { greater_than_or_equal_to: 0 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  def update_users_posts_counter
-    user.increment!(:posts_counter)
+  def update_post_counter
+    author.update(posts_counter: Post.where(author_id: author.id).count)
   end
 
-  def most_recent_comments
-    comments.order(created_at: :desc).limit(5)
+  def recent_comments
+    comments.order(created_at: :desc).limit(3)
   end
```

Comment:
- [ **OPTIONAL** ] **RECOMMENDED:** Good job fixing all issues pointed out by the previous reviewer, as the one above was not mentioned by him, I think it would be better if you update the `recent_comments` method to still showing **5** comments and not `3` for the project requirements. i.e.
```ruby
  ...
  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end
  ...
```